shapely/tests/test_ragged_array.py

```diff
@@ -86,6 +86,17 @@ def test_include_z_default():
     assert coords.shape[1] == 2
 
 
+@pytest.mark.parametrize("geom", all_types)
+def test_read_only_arrays(geom):
+    # https://github.com/shapely/shapely/pull/1744
+    typ, coords, offsets = shapely.to_ragged_array([geom, geom])
+    coords.flags.writeable = False
+    for arr in offsets:
+        arr.flags.writeable = False
```

Comment:
ah cool; I didn't know you could set a buffer as immutable so easily!

Reply:
Yeah, in numpy nothing is ever _really_ immutable, as you can always easily change that flag to True :)